packages/test-interface-criteria/src/criteria/TestOpportunityBookable.js

```diff
@@ -1,35 +1,27 @@
+const moment = require('moment');
+
 const CriteriaFutureScheduledOpportunity = require('./CriteriaFutureScheduledOpportunity')
 
 /*
   Implements https://openactive.io/test-interface#TestOpportunityBookable
 */
 
 module.exports = class TestOpportunityBookable extends CriteriaFutureScheduledOpportunity {
-  testMatch(opportunity) {
-    let {matchesCriteria, unmetCriteriaDetails} = super.testMatch(opportunity);
-
-    var id = this.getId(opportunity);
-    var type = this.getType(opportunity);
-
-    // Check for bookability
-    var bookableOffers = this.getBookableOffers(opportunity);
-    var remainingCapacity = this.getRemainingCapacity(opportunity);
-
-    if (
-      !(bookableOffers.length > 0)
-    ) {
-      matchesCriteria = false;
-      unmetCriteriaDetails.push("No bookable Offers")
-    }
+  get opportunityConstraints() {
+    return {
+      ...super.opportunityConstraints,
+      'Remaining capacity must be non-zero': opportunity => this.getRemainingCapacity(opportunity) > 0
+    };
+  }
 
-    if (
-      !(remainingCapacity > 0)
-    ) {
-      matchesCriteria = false;
-      unmetCriteriaDetails.push("No remaining capacity")
+  get offerConstraints() {
+    return {
+      ...super.offerConstraints,
+      'Must have "bookable" offer': (x, opportunity) =>
+        (Array.isArray(x.availableChannel) && x.availableChannel.includes("https://openactive.io/OpenBookingPrepayment"))
+        && x.advanceBooking != "https://openactive.io/Unavailable"
```

Comment:
```suggestion
        && x.advanceBooking !== "https://openactive.io/Unavailable"
```

though perhaps the bigger thing is that eslint should be set up for test-interface-criteria

Reply:
Good catch, and yes will do!